src/mds/Server.cc

```diff
@@ -11994,6 +12045,64 @@ bool Server::build_snap_diff(
 }
 
 #ifdef WITH_CEPHFS_NOTIFICATION
+void Server::handle_client_add_kafka_topic(const MDRequestRef &mdr) {
+  const cref_t<MClientRequest> &req = mdr->client_request;
+  KafkaTopicPayload payload;
+  if (req->get_data().length()) {
+    auto iter = req->get_data().cbegin();
+    decode(payload, iter);
+  }
+  auto m = make_message<MNotificationInfoKafkaTopic>(
+      payload.topic_name, payload.broker, payload.use_ssl, payload.user,
+      payload.password, payload.ca_location, payload.mechanism);
+  mds->send_notification_info_to_peers(m);
```

Comment:
I'd prefer to have "update-all-or-nothing" approach here. Perhaps to be implemented later though...
But IMO for now it's worth to update local setting first and proceed with populating them to peers if successul only.